app/models/user.rb

```diff
@@ -9,4 +9,44 @@ class User < ApplicationRecord
   has_many :posts
   has_many :comments, dependent: :destroy
   has_many :likes, dependent: :destroy
+
+  has_many :friendships
+  has_many :inverse_friendships, class_name: 'Friendship', foreign_key: 'friend_id'
+
+  validates :name, presence: true, length: { maximum: 50 }
+  validates :email, presence: true, length: { maximum: 50 }
+
+  def friends
+    friends_array = friendships.map do |friendship|
+      friendship.friend if friendship.confirmed == 1
+    end
+    friends_array.concat(inverse_friendships.map do |friendship|
+      friendship.user if friendship.confirmed == 1
+    end)
+    friends_array.compact
+  end
+
+  def pending_friends
+    friends_array = friendships.map do |friendship|
+      friendship.friend unless friendship.confirmed == 1
+    end
+    friends_array.compact
+  end
+
+  def friend_requests
+    friend_requests = inverse_friendships.map do |friendship|
+      friendship.user unless friendship.confirmed == 1
+    end
+    friend_requests.compact
+  end
+
+  def confirm_friend(user)
+    friendship = inverse_friendships.find { |friends| friends.user == user }
+    friendship.confirmed = 1
+    friendship.save
+  end
```

Comment:
You need to implement the front end so the final user can confirm friend requests as required on the [Pathwright](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/final_project/milestone5/accept_reject_friendship.png).